app/src/retrieve.py

```diff
@@ -6,14 +7,40 @@
 import src.adapters.db as db
 from src.db.models.document import Chunk
 
+logger = logging.getLogger(__name__)
+
 
 def retrieve(
     db_session: db.Session, embedding_model: SentenceTransformer, query: str, k: int = 5
+) -> Sequence[Chunk]:
+    logger.info(f"Retrieving context for {query!r}")
+
+    query_embedding = embedding_model.encode(query, show_progress_bar=False)
+
+    chunks = db_session.scalars(
+        select(Chunk).order_by(Chunk.mpnet_embedding.max_inner_product(query_embedding)).limit(k)
+    ).all()
+
+    for chunk in chunks:
+        logger.info(f"Retrieved: {chunk.document.name!r}")
```

Comment:
Should this be `debug`?
```suggestion
        logger.debug(f"Retrieved: {chunk.document.name!r}")
```